pkg/trace/stats/aggregation.go

```diff
@@ -99,4 +146,9 @@ func NewAggregationFromGroup(g *pb.ClientGroupedStats) Aggregation {
 			Synthetics:  g.Synthetics,
 		},
 	}
+
+	if g.PeerTags != nil {
```

Comment:
I don't think you need this `nil` check. If `g.PeerTags` is in fact nil, then `peerTagsHash` will return 0 since you do a length check at the beginning of the function. And since the zero-value for `agg.PeerTagsHash` is in fact `0`, then it won't change the value if you set `agg.PeerTagsHash = peerTagsHash(g.PeerTags)` even if `peerTagsHash` returns `0`.

So basically, I'm suggesting just setting this the same way as the rest above.

Reply:
Unless you need to check `enablePeerSvcAgg` before setting this value (which isn't happening right now, but I see in other parts of the code?)

Reply:
I think you're right. This is the first part of the client stats path where the incoming data _may_ have peer.service and `PeerTags` set. If they're nil, then the hash should be 0 as expected.

During the aggregation we need to check if we're enabled because that determines whether we should include or drop that data (and re-aggregate accordingly). In the end I don't think we have to check if we're enabled here. Let me know if that makes sense.